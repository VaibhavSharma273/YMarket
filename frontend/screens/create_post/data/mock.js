mock = [{
    "id" : 123,
    "first_name": "matthew",
    "last_name": "zhang",
    "biography": "default bio",
    "email": "matthew.zhang@yale.edu",
    "posts": [
        {
            "id": 1,
            "title" : "Chair",
            "date_posted": "1/2/2011",
            "price" : "90",
            "content" : "Lorea",
            "category" : "general",
            "postimages" : [
                { 
                    "id": 15,
                    "image_url" : "https://grandrapidschair.com/wp-content/uploads/2016/01/250_Brady_Graphite_Honey-1.jpg"
                },
                { 
                    "id": 16,
                    "image_url" : "https://grandrapidschair.com/wp-content/uploads/2016/01/250_Brady_Graphite_Honey-1.jpg"
                }
            ]
        },
        {
            "id": 2,
            "title" : "Sofa",
            "date_posted": "3/23/2011",
            "price" : "180",
            "content" : "yeehaw",
            "category" : "furniture",
            "postimages": [
                { 
                    "id": 17,
                    "image_url" : "https://www.thespruce.com/thmb/yYeyL7KSSzmtzr49xwsMPicxEi4=/2005x1504/smart/filters:no_upscale()/how-to-choose-a-sofa-color-4151743-04-b2a07f452fd3476293a75aff7e5ea17c.jpg"
                },
                { 
                    "id": 18,
                    "image_url" : "https://www.thespruce.com/thmb/yYeyL7KSSzmtzr49xwsMPicxEi4=/2005x1504/smart/filters:no_upscale()/how-to-choose-a-sofa-color-4151743-04-b2a07f452fd3476293a75aff7e5ea17c.jpg"
                }
            ]
        },
        {
            "id" : 3,
            "title" : "Mirror",
            "price" : "60",
            "content" : "Lorea",
            "category" : "general",
            "date_posted": "3/23/2011",
            "postimages": [
                { 
                    "id": 42,
                    "image_url" : "https://images.kirklands.com/is/image/Kirklands/196857_1?$tProduct$"
                }
            ]
        },
    ],
}];
export default mock;