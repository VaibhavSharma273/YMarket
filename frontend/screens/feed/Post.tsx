import React from 'react';
import { View, Image, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { text_styles } from '../../components/TextNormalize';

const Post = (props: { post: any; navigation: any; is_edit: any }) => {
    const { post, navigation } = props;

    // Render image here using the image URLs:
    const renderPostContent = () => {
        if ('postimages' in post && post.postimages.length) {
            return (
                <View style={styles.listItemBody}>
                    <Image style={styles.listItemImage} source={{ uri: post.postimages[0].image_url }} />
                </View>
            );
        }
    };

    // Actual post rendering including the image and test for the feed
    return (
        <View style={styles.listItemShadow}>
            {props.is_edit === false ? (
                <TouchableOpacity
                    style={styles.listItem}
                    onPress={() =>
                        navigation.push('ViewPost', {
                            post: post,
                        })
                    }
                >
                    {renderPostContent()}
                    <View style={styles.listItemFooter}>
                        <Text numberOfLines={2} style={[text_styles.medium, { fontWeight: '600', flex: 4 }]}>
                            {post.title}
                        </Text>
                        {post.price == null ? null : (
                            <Text numberOfLines={1} style={[text_styles.medium, { textAlign: 'right', flex: 2 }]}>
                                ${post.price}
                            </Text>
                        )}
                    </View>
                </TouchableOpacity>
            ) : (
                <TouchableOpacity
                    style={styles.listItem}
                    onPress={() =>
                        navigation.push('EditPostScreen', {
                            postId: post.id,
                            post: post,
                        })
                    }
                >
                    {renderPostContent()}
                    <View style={styles.listItemFooter}>
                        <Text numberOfLines={2} style={[text_styles.medium, { fontWeight: '600', flex: 4 }]}>
                            {post.title}
                        </Text>
                        {post.price == null ? null : (
                            <Text numberOfLines={1} style={[text_styles.medium, { textAlign: 'right', flex: 2 }]}>
                                ${post.price}
                            </Text>
                        )}
                    </View>
                </TouchableOpacity>
            )}
        </View>
    );
};

// ! Issue: Hard coded margins and padding, can cause conflict with different screen sizes
const styles = StyleSheet.create({
    listItem: {
        borderRadius: 16,
        backgroundColor: '#fff',
        marginTop: 10,
        marginBottom: 10,
        marginLeft: 20,
        marginRight: 20,
    },
    listItemShadow: {
        borderRadius: 16,
        backgroundColor: 'transparent',
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 1,
        },
        shadowOpacity: 0.22,
        shadowRadius: 2.22,
        elevation: 3,
    },
    listItemBody: {
        flex: 1,
        minHeight: 320,
    },
    listItemImage: {
        borderTopLeftRadius: 16,
        borderTopRightRadius: 16,
        aspectRatio: 1,
        flex: 1,
    },
    listItemFooter: {
        flexDirection: 'row',
        padding: 8,
        paddingLeft: 16,
    },
});

export default Post;
