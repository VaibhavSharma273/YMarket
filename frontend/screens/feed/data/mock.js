mock = [{
    "id" : 1,
    "title" : "Chair",
    "price" : "$90",
    "content": "https://grandrapidschair.com/wp-content/uploads/2016/01/250_Brady_Graphite_Honey-1.jpg"
},
{
    "id": 2,
    "title" : "Sofa",
    "price" : "$180",
    "content": "https://www.thespruce.com/thmb/yYeyL7KSSzmtzr49xwsMPicxEi4=/2005x1504/smart/filters:no_upscale()/how-to-choose-a-sofa-color-4151743-04-b2a07f452fd3476293a75aff7e5ea17c.jpg"
},
{
    "id" : 3,
    "title" : "Mirror",
    "price" : "$60",
    "content": "https://images.kirklands.com/is/image/Kirklands/196857_1?$tProduct$"
}
];

export default mock;