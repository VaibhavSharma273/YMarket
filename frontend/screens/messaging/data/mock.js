mock = [
    {
        title: 'post title',
        sender: 'other',
        receiver: 'user',
        messages: [
            {
                body: 'test message',
                thread: 19,
                sender: 'other',
                receiver: 'user',
                sent_at: '2022-04-17T19:44:24.112207Z'
            },
            {
                body: 'message history',
                thread: 19,
                sender: 'user',
                receiver: 'other',
                sent_at: '2022-04-18T19:44:24.112207Z'
            }
        ]
    }
]

export default mock;