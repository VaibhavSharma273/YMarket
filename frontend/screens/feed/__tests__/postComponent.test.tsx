import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Post from '../Post';
import renderer from 'react-test-renderer';
import mock from '../data/mock';

const createTestProps = () => ({
    navigation: {
      navigate: jest.fn()
    },
    post: mock[0],
    is_edit: false
  });

describe('Post Display component', () => {
    let component: ShallowWrapper;
    let props: any;
    beforeEach(() => {
        props = createTestProps();
        component = shallow(<Post {...props} />);
    });

    it(`renders correctly`, () => {
        expect(component.length).toBe(1);
    });
    it('matches the snapshot', () => {
        const tree = renderer.create(<Post {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
