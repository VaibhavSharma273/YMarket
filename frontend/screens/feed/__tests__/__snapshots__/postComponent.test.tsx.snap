// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Display component matches the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "elevation": 3,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "minHeight": 320,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cdn.shopify.com/s/files/1/0124/5662/4187/products/102078ba05f00ead3aec6f240ab7ebaf_46b686dd-2ac4-4d7b-9ac1-001770a055f5_1000x1000.jpg?v=1620184068",
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 8,
          "paddingLeft": 16,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 37,
            },
            Object {
              "flex": 1,
              "fontWeight": "600",
            },
          ]
        }
      >
        Chair
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 37,
            },
            Object {
              "flex": 1,
              "textAlign": "right",
            },
          ]
        }
      >
        $
        90
      </Text>
    </View>
  </View>
</View>
`;
